### PR TITLE
More utils tests

### PR DESCRIPTION
**Pull request by allenh1**
Mark certain functions with `# pragma: no cover` so that they don't get counted in the coverage results, as well as add a bit more to the license tests.

```
allenh1@hunter-laptop ~/work/superflore $ coverage run --source=superflore -m "nose" && coverage report -m
..............................
----------------------------------------------------------------------
Ran 30 tests in 1.879s

OK
Name                                               Stmts   Miss  Cover   Missing
--------------------------------------------------------------------------------
superflore/CacheManager.py                            20     13    35%   23-24, 28-33, 37-41
superflore/TempfileManager.py                         34     23    32%   27-28, 31-41, 44-57
superflore/__init__.py                                10      4    60%   5-8
superflore/docker.py                                  32     12    62%   40, 43-59
superflore/exceptions.py                              12      3    75%   18, 28, 33
superflore/generate_installers.py                     57     48    16%   32-86
superflore/generators/__init__.py                      0      0   100%
superflore/generators/bitbake/__init__.py              3      1    67%   3
superflore/generators/bitbake/gen_packages.py        115     92    20%   40-103, 110-166, 174-182, 188
superflore/generators/bitbake/ros_meta.py             27     18    33%   24-27, 30-35, 38-51, 54-55
superflore/generators/bitbake/run.py                 113     94    17%   39-44, 51-199
superflore/generators/bitbake/yocto_recipe.py        112     92    18%   43-64, 67, 70-74, 77-89, 92-96, 99-101, 104-105, 113-116, 125-177
superflore/generators/ebuild/__init__.py               3      1    67%   3
superflore/generators/ebuild/ebuild.py               135      7    95%   125-127, 131-136
superflore/generators/ebuild/gen_packages.py         151    125    17%   45-113, 118-150, 155-211, 216-233, 236, 239
superflore/generators/ebuild/metadata_xml.py          33      0   100%
superflore/generators/ebuild/overlay_instance.py      37     25    32%   27-32, 35-45, 48-64, 67-68
superflore/generators/ebuild/run.py                  157    137    13%   40-43, 47-52, 59-250
superflore/repo_instance.py                           49     34    31%   26-35, 38-46, 49-57, 63, 69, 75, 81, 84-91
superflore/rosdep_support.py                          34      2    94%   83-84
superflore/utils.py                                  109     40    63%   54-59, 108, 114, 121-124, 126, 132, 138-139, 144, 148-149, 156-177
--------------------------------------------------------------------------------
TOTAL                                               1243    771    38%
```

### CODE REVIEW

File: superflore/utils.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def get_pkg_version(distro, pkg_name):
     return maj_min_patch
 
 
-def rand_ascii_str(length=10):
+def rand_ascii_str(length=10):  # pragma: no cover
```
Comment (tfoote) [inline superflore/utils.py]: This one you could test returns only ascii characters and that the length is as expected.

File: superflore/utils.py [hunk 2]
```diff
@@ -24,23 +24,23 @@
...
     print(colored('!!!! {0}'.format(string), 'red'))
 
 
-def info(string):
+def info(string):  # pragma: no cover
     print(colored('>>>> {0}'.format(string), 'cyan'))
 
 
-def make_dir(dirname):
+def make_dir(dirname):  # pragma: no cover
```
Comment (tfoote) [inline superflore/utils.py]: This could be trivially tested in a tempdir. It would be good to add tests for the different edge cases such as the directory existing, not existing, is file, and not writable.

Comment (allenh1) [reply superflore/utils.py]: That's a good point. Will do.

File: superflore/utils.py (hunk 1 above)
Comment (allenh1) [reply superflore/utils.py]: Also a good point. Thanks.

**Comment by allenh1** [comment]
Latest report below. Getting better! I'd say the next priority is `generate_installers.py`. @tfoote what do you think?

```
allenh1@hunter-laptop ~/work/superflore $ coverage run --source=superflore -m "nose" && coverage report -m
................................
----------------------------------------------------------------------
Ran 32 tests in 5.886s

OK
Name                                               Stmts   Miss  Cover   Missing
--------------------------------------------------------------------------------
superflore/CacheManager.py                            20     13    35%   23-24, 28-33, 37-41
superflore/TempfileManager.py                         34     12    65%   32-37, 48-56
superflore/__init__.py                                10      4    60%   5-8
superflore/docker.py                                  32     12    62%   40, 43-59
superflore/exceptions.py                              12      3    75%   18, 28, 33
superflore/generate_installers.py                     57     48    16%   32-86
superflore/generators/__init__.py                      0      0   100%
superflore/generators/bitbake/__init__.py              3      1    67%   3
superflore/generators/bitbake/gen_packages.py        115     92    20%   40-103, 110-166, 174-182, 188
superflore/generators/bitbake/ros_meta.py             27     18    33%   24-27, 30-35, 38-51, 54-55
superflore/generators/bitbake/run.py                 113     94    17%   39-44, 51-199
superflore/generators/bitbake/yocto_recipe.py        112     92    18%   43-64, 67, 70-74, 77-89, 92-96, 99-101, 104-105, 113-116, 125-177
superflore/generators/ebuild/__init__.py               3      1    67%   3
superflore/generators/ebuild/ebuild.py               143      7    95%   125-127, 131-136
superflore/generators/ebuild/gen_packages.py         151    125    17%   45-113, 118-150, 155-211, 216-233, 236, 239
superflore/generators/ebuild/metadata_xml.py          33      0   100%
superflore/generators/ebuild/overlay_instance.py      37     25    32%   27-32, 35-45, 48-64, 67-68
superflore/generators/ebuild/run.py                  157    137    13%   40-43, 47-52, 59-250
superflore/repo_instance.py                           49     34    31%   26-35, 38-46, 49-57, 63, 69, 75, 81, 84-91
superflore/rosdep_support.py                          34      2    94%   83-84
superflore/utils.py                                  118     41    65%   50, 54-59, 108, 114, 121-124, 126, 132, 138-139, 144, 148-149, 156-177
--------------------------------------------------------------------------------
TOTAL                                               1260    761    40%
```
